Src/Eiffel/eiffel/byte_code/visitor/assertion_breakable_slot_strategy.e

```diff
@@ -0,0 +1,481 @@
+﻿note
+	description: "Visitor for BYTE_NODE objects which count breakpoint slots in assertions."
+
+class
+	ASSERTION_BREAKABLE_SLOT_STRATEGY
+
+inherit
+	BYTE_NODE_ITERATOR
+		redefine
+			process_assign_b,
+			process_assert_b,
+			process_inv_assert_b,
+			process_require_b,
+			process_variant_b,
+			process_elsif_b,
+			process_elsif_expression_b,
+			process_if_b,
+			process_if_expression_b,
+			process_inspect_b,
+			process_instr_call_b,
+			process_loop_b,
+			process_loop_expr_b,
+			process_nested_b,
+			process_retry_b,
+			process_reverse_b,
+			process_check_b,
+			process_invariant_b,
+			process_guard_b
+		end
+
+	SHARED_BYTE_CONTEXT
+
+feature -- Status report
+
+	breakable_slot_count (a_node: BYTE_NODE; ctx: BYTE_CONTEXT): INTEGER
+			-- Does `a_node' contain an assignment to an attribute?
+		do
+			counter := 0
+			a_node.process (Current)
+			Result := counter
+
+			print (ctx.class_type.associated_class.name)
+			print (".")
+			print (ctx.current_feature.e_feature.name_8)
+			print (" -> " + a_node.generator + ": ")
+			print (Result)
+			print ("%N")
+
+		end
+
+feature {NONE} -- Implementation: access
+
+	counter: INTEGER
+
+	increment_counter (a_node: BYTE_NODE)
+		do
+			counter := counter + 1
+			if attached {ASSERT_B} a_node as a_assert and then attached a_assert.tag as t then
+				print (" - " + a_node.generator + " %"" + t + "%" [" + counter.out + "]%N")
+			else
+				print (" - " + a_node.generator + " [" + counter.out + "]%N")
+			end
+		end
+
+	generate_melted_debugger_hook_nested
+		do
+
+		end
+
+feature -- Node processing
+
+	process_assert_b (a_node: ASSERT_B)
+			-- Process `a_node`.
+			-- (from BYTE_NODE_ITERATOR)
+			-- (export status {BYTE_NODE})
+		do
+				-- Assertion mark
+			if context.assertion_type = {ASSERT_TYPE}.in_precondition then
+				increment_counter (a_node)
+				a_node.expr.process (Current)
+			else
+				if context.assertion_type /= {ASSERT_TYPE}.in_invariant then
+						-- No hooks when it is an invariant
+					increment_counter (a_node)
+				end
+					-- Assertion byte code
+				a_node.expr.process (Current)
```

Comment:
This and the same command from another branch can be moved outside of the conditional.

The conditional will simplify to
```
				if context.assertion_type /= {ASSERT_TYPE}.in_invariant then
						-- No hooks when it is an invariant
					increment_counter (a_node)
				end
```

Alternatively, the conditional can be removed altogether by moving the call to `increment_counter` to `process_require_b`, `process_variant_b`, etc.